src/main/java/org/apache/sling/feature/cpconverter/accesscontrol/DefaultAclManager.java

```diff
@@ -150,6 +164,20 @@ public void addRepoinitExtension(@NotNull List<VaultPackageAssembler> packageAss
         }
     }
 
+    private String calculateIntermediatePath(RepoPath intermediatePath) {
+        if (enforcePrincipalBased && supportedPrincipalBasedPath!= null && !intermediatePath.startsWith(supportedPrincipalBasedPath)) {
+            RepoPath parent = supportedPrincipalBasedPath.getParent();
+            if (parent.equals(intermediatePath)) {
+                return supportedPrincipalBasedPath.toString();
+            } else {
+                String relpath = intermediatePath.toString().substring(parent.toString().length());
+                return supportedPrincipalBasedPath.toString() + relpath;
+            }
+        } else {
+            return intermediatePath.toString();
+        }
+    }
+
     private boolean aclStartsWith(RepoPath path) {
```

Comment:
this method is also called for groups, where we only wanted to create groups if a policy is located at the group-node (but not below).... the subsequent test for 'isBelow' will the cause the failure.... but IMO the correct statement here would to check if there is a acl whose access controlled path equals to the group path.
for users the starts-with is correct...

Reply:
Right, that is why the group does both:
if (aclStartsWith(group.getPath())) {
                    formatter.format("create group %s with path %s%n", group.getId(), group.getIntermediatePath());
                }
                if (aclIsBelow(group.getPath())) {
                    throw new IllegalStateException("Detected policy on subpath of group: " + group);
                }
I think that is doing what you want - just in a roundabout way, no?